rir/src/runtime/ArglistOrder.h

```diff
@@ -0,0 +1,86 @@
+#ifndef ARGLIST_ORDER_H
+#define ARGLIST_ORDER_H
+
+#include "RirRuntimeObject.h"
+
+#include <iostream>
+#include <vector>
+
+namespace rir {
+
+#pragma pack(push)
+#pragma pack(1)
+
+constexpr static size_t ARGLIST_ORDER_MAGIC = 0xaabbccdd;
+
+struct Code;
+
+struct ArglistOrder
+    : public RirRuntimeObject<ArglistOrder, ARGLIST_ORDER_MAGIC> {
+
+    using CallId = size_t;
+    using ArgIdx = size_t;
+    using CallArglistOrder = std::vector<ArgIdx>;
```

Comment:
unfortunately this will leak, since RirRuntimeObjects do not have their distructors called when the R GC collects them...

Reply:
ah, gotcha :)